docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,479 @@
+
+[!code-csharp[EvaluateModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#EvaluateModelMain "Add EvaluateModel method in Main")]
+
+
+## Use your model
+
+Now you can use your trained model to make predictions on new data.
+
+Outside of `Main()`, create a method called `UseModelForSinglePrediction()`:
+```csharp
+public static void UseModelForSinglePrediction(MLContext mlcontext, ITransformer model)
+{
+
+
+}
+```
+
+Define the method by adding the following code to `UseModelForSinglePrediction()`:
+
+[!code-csharp[PredictionEngine](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#PredictionEngine "Create Prediction Engine")]
+
+The [PredictionEngine class](xref:Microsoft.ML.PredictionEngine%602) is a convenience API, which allows you to pass a single instance of data and then perform a prediction on this single instance of data.
+
+Add the following as the next lines in the `UseModelForSinglePrediction()` method:
+
+[!code-csharp[MakeSinglePrediction](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#MakeSinglePrediction "Make a single prediction with the Prediction Engine")]
+
+Here you create an instance of `MovieRating` called `testInput` and pass it to the Prediction Engine. The [Predict()](xref:Microsoft.ML.PredictionEngine%602.Predict%2A) function makes a prediction on a single column of data (in this case on the `testInput`). 
+
+You can then use the `Score`, or the predicted rating, to determine whether you want to recommend the movie with movieId 10 to user 6. The higher the `Score`, the higher the likelihood of a user liking a particular movie. In this case, let’s say that you recommend movies with a predicted rating of > 3.5.
+
+Add the following as the next lines in the `UseModelForSinglePrediction()` method to print the results:
+
+[!code-csharp[PrintResults](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#PrintResults "Print the recommendation prediction results")]
+
+Add the following as the next line of code in the `Main()` method to call your `UseModelForSinglePrediction()` method:
+
+[!code-csharp[UseModelMain](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#UseModelMain "Add UseModelForSinglePrediction method in Main")]
+
+### Save your model
+To use your model to make predictions in end-user applications, you must first save the model.
+
+Outside of `Main()`, create a method called `SaveModel()`:
+```csharp
+public static void SaveModel(MLContext mlcontext, ITransformer model){
+{
+
+
+}
+```
+
+Define the method by adding the following code to `SaveModel()`:
+
+[!code-csharp[SaveModel](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#SaveModel "Save the model to a zip file")]
+
+This saves the model to a .zip file, which can then be loaded into and used in your other .NET applications.
```

Comment:
Move this explanation to right before "Define the method by adding the following code to `SaveModel()`:".